### ISSUE TITLE
Validator using RegEx pattern does not work

### ISSUE THREAD

**Issue by aspetko** [angular/angular#13704]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
In a form I am validating the values of eMail address and internet site. The Pattern did not work in either the HTML template, nor the typescript file. 
I ended with a pattern like 
...
email: ['', Validators.pattern('^[.a-zäöüßÄÖÜA-Z0-9~]+@[a-zäöüßÄÖÜA-Z0-9~]+\.[a-zäöüßÄÖÜA-Z0-9]+$')],
internetAdresse: ['', Validators.pattern('^([hH][tT][tT][pP][sS]?:\/\/)?([a-zäöüßÄÖÜA-Z0-9]+(-[a-zA-ZäöüßÄÖÜ0-9]+)*\.)+[a-zäöüßÄÖÜA-Z]{2,}$')],
...
but it is still not working correct.

The following errors occurred:
1. I can enter blanks in the internet address. The email address is working.
2. I can insert one or more than one @ sign as the internet address. The email adress working in this case again.
3. If I enter in the pattern \s instead of   [a-zäöüßÄÖÜA-Z0-9] it is not disallowing blanks

**Expected behavior**
Set the form to invalid, when a unspecified character is entered. 
**Minimal reproduction of the problem with instructions**
Create a form, assign the pattern to a textfield. Now try to enter characters as described above.
**What is the motivation / use case for changing the behavior?**
We (German Government) are using an Angular 2 app to save the citizens time by allowing them to enter forms over the internet. Validation is therefor required. 

**Please tell us about your environment:**
Linux, Windows
Webstorm
webpack
weblogic cluster
2 core 8 GB RAM

* **Angular version:** 2.0.2

* **Browser:** [Chrome current 64-bit | Firefox ESR 45.2 | IE 11 ]
 
* **Language:** [ TypeScript 2.1.0-dev.20160915 ]

* **Node (for AoT issues):** `node --version` =   6.2.0

**Comment by DzmitryShylovich** [comment]
Pls add a repro in plunkr

**Comment by DzmitryShylovich** [comment]
> Validators.pattern('^[.a-zäöüßÄÖÜA-Z0-9~]+@[a-zäöüßÄÖÜA-Z0-9~]+.[a-zäöüßÄÖÜA-Z0-9]+$')]

You don't need to add `^` https://github.com/angular/angular/blob/master/modules/%40angular/forms/src/validators.ts#L109

**Comment by tytskiy** [comment]
@aspetko if you pass a string it will construct RegExp with start-end symbols as @DzmitryShylovich shows.
So you need to remove both `^` at the beginning and `$` at the end.

If you need more flexible behavior, then pass raw RegExp object rather that simple string.

**Comment by aspetko** [comment]
Thank you for your fast response. 

@DzmitryShylovich , @tytskiy I have removed the ^ and $ from the pattern. After recompiling, I was still able to reproduce the behavior.

@DzmitryShylovich I would love to create a plunkr repo, but I can't do it before 02.01.2017.
@tytskiy  I have tried it like this:
email: ['', Validators.pattern(new RegExp(/^[.a-zäöüßÄÖÜA-Z0-9~]+@[a-zäöüßÄÖÜA-Z0-9~]+\.[a-zäöüßÄÖÜA-Z0-9]+$/))],
internetAdresse: ['', Validators.pattern(new RegExp(/^([hH][tT][tT][pP][sS]?:\/\/)?([a-zäöüßÄÖÜA-Z0-9]+(-[a-zA-ZäöüßÄÖÜ0-9]+)*\.)+[a-zäöüßÄÖÜA-Z]{2,}$/))],

It is even whorse.

**Comment by DzmitryShylovich** [comment]
`Validators.pattern(/^[.a-zäöüßÄÖÜA-Z0-9~]+@[a-zäöüßÄÖÜA-Z0-9~]+.[a-zäöüßÄÖÜA-Z0-9]+$/]`

**Comment by tytskiy** [comment]
@aspetko sorry for confusion, i missed you use `Angular version: 2.0.2` which does not support raw RegExp. It only support string. And RegExp was added later.

Try `Validators.pattern('[.a-zäöüßÄÖÜA-Z0-9~]+@[a-zäöüßÄÖÜA-Z0-9~]+.[a-zäöüßÄÖÜA-Z0-9]+')`

If it doesn't work you can create custom validator function.

As for the first problem:
>1. I can enter blanks in the internet address. The email address is working.

It works as expected as `Validator.pattern` is not supposed to validate against empty value. You need to add `Validators.required` before your `Validator.pattern`

Please refer to docs: 
```
    this.heroForm = this.fb.group({
      'name': [this.hero.name, [
          Validators.required,
          Validators.minLength(4),
          Validators.maxLength(24),
          forbiddenNameValidator(/bob/i)
        ]
      ],
      'alterEgo': [this.hero.alterEgo],
      'power':    [this.hero.power, Validators.required]
    });
```
https://angular.io/docs/ts/latest/cookbook/form-validation.html

**Comment by aspetko** [comment]
@tytskyi Don't worry, everybody does mistakes. Even I do ;)

I tried implementing it without using regex, but it was very complicated. But the good news is, that I took than my regex, and placed it in the HTML form using the pattern keyword. 

It worked, except for beginning and ending blanks. But these I can easily remove using trim.

Unfortunatly, I don't have the time to create a plunkr example. We are going to production in six days (!).

After this version is live, we can see if we can get rid of the nasty bug.

Cheers Alexander
